lib/rubocop/cop/lint/assignment_in_condition.rb

```diff
@@ -53,7 +59,11 @@ def on_if(node)
             next :skip_children if skip_children?(asgn_node)
             next if allowed_construct?(asgn_node)
 
-            add_offense(asgn_node.loc.operator)
+            add_offense(asgn_node.loc.operator) do |corrector|
+              next unless safe_assignment_allowed?
+
+              corrector.replace(asgn_node, "(#{asgn_node.source})")
```

Comment:
That's my two cents:

```suggestion
              corrector.wrap(asgn_node, '(', ')')
```